docs/features/records.md

```diff
@@ -21,78 +21,144 @@ This new form is called a *caller-receiver default-argument*, and is allowed onl
 
 When an argument is omitted from an invocation of a function member for a corresponding optional parameter with a *caller-receiver default-argument*, the value of the receiver's member is implicitly passed. 
 
+> **Design Notes**: the main reason for the caller-receiver parameter is to support the *with-expression*. The idea is that you can declare a method like this
+> ```cs
+> class Point
+> {
+>     public readonly int X;
+>     public readonly int Y;
+>     public Point With(int x = this.X, int y = this.Y) => new Point(x, y);
+>     // etc
+> }
+> ```
+> and then use it like this
+> ```cs
+>     Point p = new Point(3, 4);
+>     p = p.With(x: 1);
+> ```
+> To create a new `Point` just like an existing `Point` but with the value of `X` changed.
+> 
+> It is an open question whether or not the syntactic form of the *with-expression* is worth adding once we have support for caller-receiver parameters, so it is possible we would do this *instead of* rather than *in addition to* the *with-expression*.
+
 - [ ] **Open issue**: What is the order in which a *caller-receiver default-argument* is evaluated with respect to other arguments? Should we say that it is unspecified?
```

Comment:
Is there a statement about how many times and when the default member access is evaluated?